### PR TITLE
Add a POST method for definitions

### PR DESCRIPTION
**Pull request by AntoineCezar**
Lettuce tests fail but I don't know if it's still used. Is it?

### CODE REVIEW

**Comment by leplatrem** [comment]
Great :) Activity !

Unfortunately many lettuce tests were not ported to unittest yet #31, so it's still relevant to keep them green ;)

Regarding the name generation, if you generate a uuid, it would be simpler to rely on couchdb, like it's done for data items : https://github.com/spiral-project/daybed/blob/master/daybed/views/data.py#L45-L49
However, it becomes very interesting if we generate the model id from the model title (a-la django slugify + loop for uniqueness)

**Comment by AntoineCezar** [comment]
I've done it this way to allow other generation method. The originale use case wanted to generate human readable names.
Anyway there is still a problem : I've forgoten to return ne generated name.

**Comment by AntoineCezar** [comment]
I've returned the name of the definition. I think returning the definition as stated in the issue wouldn't be consistant

**Comment by leplatrem** [comment]
I'm no expert about REST conventions, but instead of returning the complete definition we could just add a redirect to the response like it's done on data post:
https://github.com/spiral-project/daybed/blob/master/daybed/views/data.py#L46-L48

**Comment by AntoineCezar** [comment]
If we redirect. How to send the token ?
The client must actualy read the response, save the token then may follow the redirect. Sounds weird to me

**Comment by AntoineCezar** [comment]
opened a ticket #61

**Comment by AntoineCezar** [comment]
I should not rebase late on nigth after vine (Macvin du Jura) since I don't master rebase... Now I must find a way to clean the history

**Comment by almet** [comment]
aha! Wine coding is always the best way to go.

Yeah, we probably better handle the tokens later on and keep things simple for now.

File: daybed/name_generators/random.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from uuid import uuid4
+
+
+class RandomNameGenerator(object):
```
Comment (almet) [inline daybed/name_generators/random.py:4]: Oh, I was thinking about something which creates better names! Like that: https://github.com/seejohnrun/haste-server/blob/master/lib/key_generators/phonetic.js

Or something ala heroku, that's far better than the uuid4!

**Comment by almet** [comment]
Otherwise sounds good to me!

**Comment by leplatrem** [comment]
Ok, then we can return the full object and its URL as attribute instead of redirect, no ?

In the contrary, I would like to keep tokens on definitions until we find a proper way to implement ACLs :)

File: daybed/name_generators/random.py (hunk 1 above)
Comment (AntoineCezar) [inline daybed/name_generators/random.py:4]: I've made the key generator plugable on purpose. UUIDs works for me but other default generators are welcomed.
Should I rename the package "random-uuid" ?

**Comment by AntoineCezar** [comment]
Created the dev branch

**Comment by leplatrem** [comment]
Is that stuck until #61 is fixed ?

**Comment by almet** [comment]
We've made a bunch of changes in the dev branch; I don' think that's too much hassle to port your changes there. I think we're ready for this.

**Comment by almet** [comment]
Cherry-picked and updated with our latest changes. see e56cd08c7678d32f1472cf3323c17ed15fb1c594 and 85827ee32e6702d08045035d8fd3a95018e7b3f4

**Pull request closed without merging**